File: Data Structures & Algorithms/subsets-ii/submission-3.cpp
class Solution {
public:
    set<vector<int>> s ;
    void dfs(int i , vector<int>& current , int n , vector<int>& nums ){
        vector<int> copy = current;
        sort(copy.begin(),copy.end());
        s.insert(copy);
        if(i==n) return ;
        current.push_back(nums[i]);
        dfs(i+1,current,n,nums);
        current.pop_back();
        dfs(i+1,current,n,nums);
    }
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<int> v ;
        dfs(0, v ,nums.size(),nums);
        return vector<vector<int>>(s.begin(),s.end());
    }
};

File: Data Structures & Algorithms/jump-game/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    canJump(nums) {
        let n= nums.length ;
         const dfs = (i)=>{
            if(i==n-1) return true ;
            else {
                if(nums[i]==0) return false ;
                else{
                    for(let j=1 ; j<=nums[i];j++){
                        if(i+j< n && dfs(i+j)) return true ;
                    }
                }
            }
            return false
         }
         return dfs(0);
    }
}
